sdk/monitor/azure-monitor-query/azure/monitor/query/_helpers.py

```diff
@@ -41,13 +41,21 @@ def get_metrics_authentication_policy(
 
     raise TypeError("Unsupported credential")
 
-def process_error(exception):
-    raise_error = HttpResponseError
-    raise raise_error(message=exception.message, response=exception.response)
-
-def order_results(request_order, mapping, obj):
+def order_results(request_order, mapping, obj, err, allow_partial_errors=False):
     ordered = [mapping[id] for id in request_order]
-    return [obj._from_generated(rsp) for rsp in ordered] # pylint: disable=protected-access
+    results = []
+    for item in ordered:
+        if not item.body.error:
+            results.append(obj._from_generated(item.body)) # pylint: disable=protected-access
+        else:
+            error = item.body.error
+            if allow_partial_errors and error.code == 'PartialError':
+                res = obj._from_generated(item.body) # pylint: disable=protected-access
+                res.partial_error = err._from_generated(error) # pylint: disable=protected-access
+                results.append(res)
+            else:
```

Comment:
Shouldn't we raise if allow_partial_errors = False and there is error?

Reply:
no - in case of batch, we return the error object instead of the result object

of course, we raise when something fatal happens (like auth error for instance)